### DISCUSSION TITLE
refactor the test that checks the mutation run in the correct folder

### DISCUSSION THREAD

**Discussion by alessio-perugini** [go-gremlins/gremlins#172] [Development]
In #171 I proposed a version of the test that I think adds up unneeded complexity.
You can find an alternative here: https://github.com/alessio-perugini/gremlins/pull/1/files

Adding the cmd directly to `commandHolder` and checking the Dir value, which will be changed by the actual implementation, feels more right to me. 

If we choose to implement the proposed version that you can find in the link above, we might want to remove the changes I made to the `dealerStub`. (Even though retaining the map in the `dealerStub` may become handy in the future)

**Comment by k3rn31** [comment]
I was thinking to do, more or less, the same, but I didn't want to bother you with too much changes in the PR :)

Not sure if we need all the changes in the dealerStub, maybe we can postpone when we actually need them.

**Reply by alessio-perugini** [reply]
@k3rn31 agree to not bring the changes of the dealerStub. If you don't mind I'll open a PR to refactor that test 😁

**Reply by k3rn31** [reply]
Sure, go ahead please!

**Reply by k3rn31** [reply]
Btw, we completed the issues for 0.4.0, which I was postponing due to the random behaviour. Now that we have your great fix, we're ready to go!
I'll wait for you to refactor this test, then, as soon as your PR is merged, we're set for the 0.4 release.

**Reply by alessio-perugini** [reply]
:rocket: :full_moon: Thank you for being so responsive. I had a lot of fun hunting down that bug.

**Reply by k3rn31** [reply]
You did a great job!
